util/src/main/scala/scala/scalanative/util/StringUtils.scala

```diff
@@ -0,0 +1,62 @@
+package scala.scalanative.util
+
+import scala.StringContext.InvalidEscapeException
+import scala.annotation.tailrec
+
+object StringUtils {
+
+  /**
+   * Custom implementation of StringContext.processEscapes which also parses hex values
+   * @param str UTF-8 encoded input string optionally containing literal escapes and hex values
+   * @return UTF-8 representation of escaped ByteString
+   */
+  def processEscapes(str: String): String = {
```

Comment:
Since this method really parses a *byte string*, it should directly return an `Array[Byte]`. It should not try to encode the byte string into a String, only to then re-decode that String into a byte string by calling `getBytes("UTF-8")`. As such, it should use an `Array.newBuilder[Byte]` instead of a `StringBuilder`.

This will avoid many indirections, and would allow to parse byte strings that are not valid UTF-8 sequences (but could be valid in other interpretations, which we must not prevent).